src/test/java/io/confluent/connect/elasticsearch/ElasticsearchWriterTest.java

```diff
@@ -298,6 +303,96 @@ public void testBytes() throws Exception {
     verifySearchResults(records, ignoreKey, ignoreSchema);
   }
 
+  @Test
+  public void testIgnoreNullValue() throws Exception {
+    final boolean ignoreKey = false;
```

Comment:
At least with `ignoreKey` and `ignoreSchema`, all the other tests have those local variables declared final.

Reply:
Up to you when to comply with what's there for the sake of uniformity and when to diverge or fix things to follow our code style. But we don't follow the pattern of declaring everything `final` by default in connectors, at least for now.

Reply:
I agree that final variables are less useful in test cases, except where they're needed in Java 7 to pass into an anonymous inner class/function.

Reply:
There is a lot of value in keeping test cases brief and terse. Since it is a commonly recurring pattern in these tests, a better way might be to predefine constants for each state of `ignoreKey` and `ignoreSchema`, and then plug them into your test. For example, 

```
  private static final boolean ENFORCE_SCHEMA = true;
  private static final boolean IGNORE_SCHEMA = false;
```

and then call initWriter as follows: 

```
  initWriter(client, ENFORCE_KEY, IGNORE_SCHEMA, BehaviorOnNullValues.IGNORE);
```

Reply:
I'm not opposed to rewriting the tests to reflect our code style and/or to make them brief, readable, etc., but wouldn't it be better to address that in a separate PR? Don't want to clutter this one up and/or make it take longer than necessary, since changes in style aren't required for the new feature to work properly. Only asking because I feel like this is a bit of a can of worms and might take longer than anticipated--code style is easy to have disagreements over 😉

Reply:
Handling these as a separate issue/PR might be fine if this was non-test related code or was cleaning up existing code. However, this PR is adding additional tests that we're commenting on, and I'd argue that this goes beyond style and ventures into good coding practices. I don't think it's appropriate to forgo this an incur the debt when it's easy enough to fix now.